internal/config/config_test.go

```diff
@@ -312,8 +312,15 @@ func TestInit(t *testing.T) {
 				callbackPhase = 2
 				phaseMu.Unlock()
 
-				err = os.WriteFile(filepath.Join(configDir, prefix+".yaml"), []byte(tc.changeConfigWith), 0600)
-				require.NoError(t, err, "Setup: failed to write initial config file")
+				// fsnotify docs states that truncating a file will also trigger a Write event, which means that the callback can be triggered
+				// before the file contents are properly written. In order to avoid that, we must manually open the file and write the new content
+				// to it to avoid truncating the file and triggered the event too early.
+				configFile, err := os.OpenFile(filepath.Join(configDir, prefix+".yaml"), os.O_RDWR|os.O_CREATE, os.ModePerm)
+				require.NoError(t, err, "Setup: failed to open config file")
+				_, err = configFile.WriteString(tc.changeConfigWith)
```

Comment:
If write faiols here, the file is not closed, You still need to defer the Close after the first require.NoError in addition to the manual close.